schemaregtistry.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/sirupsen/logrus"
+)
+
+type SchemaRegistrySubject struct {
+	ID      uint32 `json:"id"`
+	Version uint32 `json:"version"`
+	Subject string `json:"subject"`
+	Schema  string `json:"schema"`
+}
+
+func getLatestSubject(registryUrl string, subjectName string) (*SchemaRegistrySubject, error) {
+	url, err := url.Parse(registryUrl)
+	if err != nil {
+		logrus.WithError(err).Error("Couldn't parse registry URL.")
+		return nil, err
```

Comment:
Rather than logging an error message and then returning the original, we should wrap the error using errors.Wrap:
```go
    return nil, errors.Wrap(err, "Couldn't parse registry URL.")
```